loopdb/sqlc/migrations/000003_sweep_batcher.up.sql

```diff
@@ -0,0 +1,55 @@
+-- sweep_batches stores the on-going swaps that are batched together.
+CREATE TABLE sweep_batches (
+        -- id is the primary key of the batch.
+        id BLOB PRIMARY KEY,
+
+        -- batch_state is the state of the batch.
+        batch_state TEXT NOT NULL,
```

Comment:
There's 3 states:
`open`: when a batch is able to add new sweeps to it
`closed`: when a batch was just spent and now its monitoring for conf height
`confirmed`: when a batch was spent and the spending tx has surpassed conf height

Reply:
I still don't really see the benefit of having a "closed" state. Besides making the code more complex is there anything we gain from it?

Reply:
It signals that the batch tx entered a block, but hasn't reached the required conf depth yet. On restart it is also used to indicate that the batch should be resumed in order to continue monitoring for confs.

Reply:
I understand what you meant. But instead of a new state, wouldn't it be simpler to just assume that a confirmed tx needs to be monitored unless N blocks old? So on restart we'd SQL select those batches that are confirmed but <= N blocks old.